api/turing/config/config.go

```diff
@@ -159,6 +159,10 @@ type ImageBuildingConfig struct {
 	BaseImageRef map[string]string `validate:"required"`
 	// KanikoConfig contains the configuration related to the kaniko executor image builder.
 	KanikoConfig KanikoConfig `validate:"required"`
+	// TolerationName allow the scheduler to schedule image building jobs with the matching name
+	TolerationName *string
```

Comment:
This is somewhat different from how Merlin saves this particular config (https://github.com/gojek/merlin/blob/83199a79831687bb9b230f5a3029716f05cd448e/api/config/config.go#L156) but I decided to store it like this because it's how we currently storing tolerations in the `SparkAppConfig` here https://github.com/caraml-dev/turing/blob/main/api/turing/config/config.go#L197.